src/clojure/nrepl/middleware/lookup.clj

```diff
@@ -33,7 +33,7 @@
           lookup-fn (or (and lookup-fn (misc/requiring-resolve (symbol lookup-fn))) *lookup-fn*)]
       (response-for msg {:status :done :info (lookup-fn ns sym)}))
     (catch Exception _e
-      (response-for msg {:status #{:done :lookup-error}}))))
+      (response-for msg {:status #{:done :lookup-error :namespace-not-found}}))))
```

Comment:
As this is a catch all, this can be some other exception as well (e.g. something might blow up in the `lookup-fn`). For proper granularity this should be contingent on the ns not being present, but I guess this would mean adding some additional check.